### PR TITLE
fix:Accelerometer Crash Fixed and Updated changes with current developmen…

### LINKED ISSUE #1465: Crashes when recording data on Accelerometer instrument and run Power Source
**Actual Behaviour**

It crashes when I record data in the accelerometer and run power source.
Doesn't work graphics at "sensors" and multimeter (nothing happens when I run it)

**Expected Behaviour**
Multimeter should work. The application should not crash.

**Would you like to work on the issue?**

Yes
screenshot_pslab_20181210-103438

### PR DESCRIPTION
**Pull request by ho-dor**
Fixes #1465 

**Changes**: Accelerometer Data, Settings Fragment added and Adapter Removed

**Screenshots for the changes**:

<table>
<tr>
<td></td>
<td></td>
<td></td>
</tr>
</table>

**Checklist**: 

- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [ ] No modifications are done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2796229/app-debug.zip)

### CODE REVIEW

**Comment by ho-dor** [comment]
@abhinavraj23 @CloudyPadmal  I have created this new PR because I was not able to pull the latest changes to my local repo after the "added barometer and images" commit . This PR is updated with the latest changes after Barometer was added and it is successfully building and running but can you please help me understand he travis CI error here . Thank You.

File: gradlew.bat [hunk 1]
```diff
@@ -1,90 +1,90 @@
-@if "%DEBUG%" == "" @echo off
```
Comment (CloudyPadmal) [inline gradlew.bat]: You have edited these files. That's why gradle build is failing in Travis

Comment (ho-dor) [reply gradlew.bat]: @CloudyPadmal can you please help me how to solve this because there is no difference between the gradlew.bat file before and after this commit as you can see in the diff also I never made any changes in this file while fixing this error . Thank You.

**Comment by ho-dor** [comment]
@abhinavraj23 @CloudyPadmal  please review . Thank You

**Comment by ho-dor** [comment]
Some are new files like AccelerometerFragment but in some files there are changes like in SensorLoggerListAdapter.

File: README.md [hunk 2]
```diff
@@ -7,7 +7,6 @@
 [![Gitter](https://badges.gitter.im/fossasia/pslab.svg)](https://gitter.im/fossasia/pslab?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
 [![Codacy Badge](https://api.codacy.com/project/badge/Grade/dd728d91bb5743ff916c16c1251f8dd5)](https://www.codacy.com/app/praveenkumar103/pslab-android?utm_source=github.com&amp;utm_medium=referral&amp;utm_content=fossasia/pslab-android&amp;utm_campaign=Badge_Grade)
 [![Mailing List](https://img.shields.io/badge/Mailing%20List-FOSSASIA-blue.svg)](mailto:pslab-fossasia@googlegroups.com)
-[![Twitter Follow](https://img.shields.io/twitter/follow/pslabio.svg?style=social&label=Follow&maxAge=2592000?style=flat-square)](https://twitter.com/pslabio)
```
Comment (CloudyPadmal) [inline README.md]: Why are these removed?

File: README.md [hunk 3]
```diff
@@ -129,13 +128,6 @@ To debug over Wi-Fi: http://blog.fossasia.org/android-app-debugging-over-wifi-fo
 Note :
 1. If you built your own hardware, change VendorID and/or ProductID in [CommunicationHandler.java](https://github.com/fossasia/pslab-android/blob/master/app/src/main/java/org/fossasia/pslab/communication/CommunicationHandler.java)
 
-### Permissions Required
```
Comment (CloudyPadmal) [inline README.md]: These also

**Comment by ho-dor** [comment]
@abhinavraj23 all the changes in this PR are made by me . I have created the new files which were not present before like SettingsFragment and DataFragment for accelerometer to incorporate PSLabSensor class for accelerometer like in Barometer and LuxMeter .
Which files are you concerned about ?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1094,4 +1095,7 @@
     <string name="recorder">Recorder</string>
     <string name="save_graph">Save Graph</string>
 
+    <!-- TODO: Remove or change this placeholder text -->
```
Comment (CloudyPadmal) [inline app/src/main/res/values/strings.xml]: This section can be removed

**Comment by ho-dor** [comment]
@CloudyPadmal didn't see the changes you requested . Sorry for the delay.

**Comment by abhinavraj23** [comment]
@ho-dor The changes requested by @CloudyPadmal is still not done, do it and then it is good to go 👍

**Comment by ho-dor** [comment]
@abhinavraj23 Out of the two string resources asked to remove , I removed one which was not in use but the other one is used at multiple places 'save graph' . So I did not remove it .

**Comment by ho-dor** [comment]
@CloudyPadmal please review . There was an issue with the sensor bottomsheet commit which resulted in crash of sensors so I fixed that in this PR and hence a couple of extra files have been added to the commit like sensor_main.xml.

**Comment by ho-dor** [comment]
@abhinavraj23 please review.

**Comment by cweitat** [comment]
@ho-dor please fix codacy

**Comment by ho-dor** [comment]
@CloudyPadmal is anything wrong in this now ? thanks .

**Comment by mariobehling** [comment]
Thank you! Please resolve conflicts.

**Comment by cweitat** [comment]
Check Codacy

**Comment by ho-dor** [comment]
please review @CloudyPadmal @abhinavraj23 @cweitat @mariobehling

File: app/src/main/res/values/strings.xml [hunk 5]
```diff
@@ -1091,7 +1092,9 @@
     GND. GND is meant for Ground and any of the PSLab device GND pins can be used since they are common.\n\n
     \u2022 Select sensor by going to the Configure tab from the bottom navigation bar and choose BHT-1750 in the drop down menu under Select Sensor.\n</string>
     <string name="no_data_fetched">No Data Fetched</string>
-    <string name="recorder">Recorder</string>
-    <string name="save_graph">Save Graph</string>
+    <!--<string name="save_graph">Save Graph</string>-->
+
+    <!-- TODO: Remove or change this placeholder text -->
+    <string name="hello_blank_fragment">Hello blank fragment</string>
```
Comment (harsh-2711) [inline app/src/main/res/values/strings.xml]: Don't add string resources at the bottom of the file

File: app/src/main/res/menu/sensor_data_log_menu.xml [hunk 6]
```diff
@@ -19,12 +19,12 @@
         android:orderInCategory="3"
         android:title="@string/play"
         app:showAsAction="ifRoom" />
-    <item
-        android:id="@+id/save_graph"
-        android:icon="@drawable/menu_icon_save"
-        android:orderInCategory="4"
-        android:title="@string/save_graph"
-        app:showAsAction="ifRoom" />
+    <!--<item-->
```
Comment (harsh-2711) [inline app/src/main/res/menu/sensor_data_log_menu.xml]: Remove commented code if not needed

File: app/src/main/res/menu/accelerometer_data_log_menu.xml [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        app:showAsAction="ifRoom" />
+    <item
+        android:id="@+id/show_guide"
+        android:title="@string/show_guide"
+        app:showAsAction="never" />
+    <item
+        android:id="@+id/show_logged_data"
+        android:title="@string/show_logged_data"
+        app:showAsAction="never" />
+    <!--<item
```
Comment (harsh-2711) [inline app/src/main/res/menu/accelerometer_data_log_menu.xml]: Remove unused code

File: app/src/main/res/layout/sensor_main.xml [hunk 8]
```diff
@@ -65,14 +65,16 @@
             android:layout_width="match_parent"
             android:layout_height="match_parent"
             android:layout_marginTop="@dimen/sensor_params_top" />
+
```
Comment (harsh-2711) [inline app/src/main/res/layout/sensor_main.xml]: Don't add extra lines here

File: app/src/main/res/layout/sensor_main.xml [hunk 9]
```diff
@@ -65,14 +65,16 @@
...
     </LinearLayout>
 
     <View
         android:id="@+id/shadow"
         android:layout_width="match_parent"
         android:layout_height="match_parent"
-        android:background="@color/black"
-        android:alpha="0"/>
+        android:alpha="0"
```
Comment (harsh-2711) [inline app/src/main/res/layout/sensor_main.xml]: Do we need this?

File: app/src/main/java/io/pslab/models/PSLabSensor.java [hunk 10]
```diff
@@ -349,16 +358,19 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
-                displayLogLocationOnSnackBar();
-                if (getSensorFragment() instanceof LuxMeterDataFragment) {
-                    ((LuxMeterDataFragment) getSupportFragmentManager()
-                            .findFragmentByTag(getSensorName())).saveGraph();
-                } else if (getSensorFragment() instanceof BaroMeterDataFragment) {
-                    ((BaroMeterDataFragment) getSupportFragmentManager()
-                            .findFragmentByTag(getSensorName())).saveGraph();
-                }
-                break;
+//            case R.id.save_graph:
```
Comment (harsh-2711) [inline app/src/main/java/io/pslab/models/PSLabSensor.java]: Remove this part

File: app/src/main/java/io/pslab/adapters/SensorLoggerListAdapter.java [hunk 11]
```diff
@@ -182,6 +194,24 @@ private void populateMapData(SensorDataBlock block) {
                 }
             }
             setMapDataToIntent(array);
+        } else if (block.getSensorType().equalsIgnoreCase(PSLabSensor.ACCELEROMETER)) {
+            RealmResults<AccelerometerData> data = LocalDataLog.with().getBlockOfAccelerometerRecords(block.getBlock());
+            JSONArray array = new JSONArray();
+            for (AccelerometerData d : data) {
+                try {
+                    JSONObject i = new JSONObject();
```
Comment (harsh-2711) [inline app/src/main/java/io/pslab/adapters/SensorLoggerListAdapter.java]: Please give relevant variable names

File: app/src/main/java/io/pslab/activity/DataLoggerActivity.java [hunk 12]
```diff
@@ -58,9 +58,15 @@ protected void onCreate(Bundle savedInstanceState) {
...
                 break;
+
+            case "Accelerometer":
+                getSupportActionBar().setTitle(caller);
+                categoryData = LocalDataLog.with().getTypeOfSensorBlocks(getString(R.string.accelerometer));
+
             case "Multimeter":
                 getSupportActionBar().setTitle(caller);
                 categoryData = LocalDataLog.with().getTypeOfSensorBlocks(getString(R.string.multimeter));
+
```
Comment (harsh-2711) [inline app/src/main/java/io/pslab/activity/DataLoggerActivity.java]: Don't change code added by other developers

File: app/src/main/java/io/pslab/adapters/SensorLoggerListAdapter.java (hunk 11 above)
Comment (ho-dor) [reply app/src/main/java/io/pslab/adapters/SensorLoggerListAdapter.java]: I followed the convention that was used in Luxmeter and Barometer.

**Comment by ho-dor** [comment]
@harsh-2711 @abhinavraj23 @CloudyPadmal @mariobehling please review

**Comment by ho-dor** [comment]
@mariobehling this has been due for long . Crash in accelerometer is solved by this.

**Comment by CloudyPadmal** [comment]
We'll merge this as a WIP and move forward from there. Resolve the conflict to get started. Thanks.

**Comment by ho-dor** [comment]
@CloudyPadmal please check. Thanks

**Comment by neel1998** [comment]
@ho-dor  nice work. But i have some suggestions. There is lots of redundant code since you are doing exactly same thing for all 3 axis. I would suggest make a general fragment for axis and then iterate over list of such fragments, It would make the code a lot cleaner. I have recently implemented gyroscope is similar way. Kindly take a look at it.

Also for the accelerometer, change the format of stored data, add separate column for acceleration in X,Y and Z axis. And show data of all these 3 for an instance in same row. Again you can refer to my implementation of gyroscope. Cheers!

**Comment by ho-dor** [comment]
Surely , I will take a look once your PR gets reviewed by the core team .
If they find the implementation correct , I will incorporate it in
accelerometer .Cheers .

**Comment by ho-dor** [comment]
@CloudyPadmal @abhinavraj23 any update for this PR ?

**Comment by neel1998** [comment]
@Avjeet  can you please look at my Pr for gyroscope implementation first before merging this. There is lots of redundant code(same for all 3 axis) in this implementation, which will make it hard to debug. I think this can surely be improvised. Thanks

**Comment by ho-dor** [comment]
@Avjeet I will do it asap . Thanks

**Comment by Avjeet** [comment]
@ho-dor is there any update on this PR?

**Comment by neel1998** [comment]
@Avjeet  did you take a look at my previous suggestion. I really feel it would be beneficial to implement this feature in a better way.

**Comment by ho-dor** [comment]
@Avjeet I am working on it.

**Comment by ho-dor** [comment]
@Avjeet I will make an updated PR in few days as my end semester exams are
going on . Thanks.

**Comment by ho-dor** [comment]
@Avjeet please review . I have updated the apk file

File: app/src/main/java/io/pslab/fragment/AccelerometerDataFragment.java [hunk 13]
```diff
@@ -0,0 +1,921 @@
...
+        }
+        graphTimer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                handler.post(new Runnable() {
+                    @Override
+                    public void run() {
+                        try {
+                            boolean playComplete = false;
+                            AccelerometerData d = recordedAccelerometerArray.get(turns);
```
[the remark this answers is not in the document]
Comment (ho-dor) [reply app/src/main/java/io/pslab/fragment/AccelerometerDataFragment.java]: I have followed the conventions used for Lux meter and Barometer

**Comment by ho-dor** [comment]
@abhinavraj23 Since there are a lot of new files in this commit , it will become hard to pass the travis if I break the commits into small pieces.
